src/AddressRegistry.Projections.Syndication/LinkedFeedProjectionManager.cs

```diff
@@ -0,0 +1,31 @@
+namespace AddressRegistry.Projections.Syndication
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Autofac.Features.OwnedInstances;
+    using Be.Vlaanderen.Basisregisters.ProjectionHandling.Runner;
+
+    public class LinkedFeedProjectionManager<TContext> where TContext : RunnerDbContext<TContext>
+    {
+        private readonly IEnumerable<ILinkedFeedProjectionRunner<TContext>> _linkedFeedProjectionRunners;
+
+        public LinkedFeedProjectionManager(IEnumerable<ILinkedFeedProjectionRunner<TContext>> linkedFeedProjectionRunners)
+        {
+            _linkedFeedProjectionRunners = linkedFeedProjectionRunners;
+        }
+
+        public async Task CatchUpAsync(Func<Owned<TContext>> context, CancellationToken cancellationToken)
+        {
+            while (true)
+            {
+                foreach (var linkedFeedProjectionRunner in _linkedFeedProjectionRunners)
+                    await linkedFeedProjectionRunner.CatchUpAsync(context, cancellationToken);
+
+                Thread.Sleep(10000 / _linkedFeedProjectionRunners.Count());
```

Comment:
Why are we sleeping?
Why are we in a while loop?

This part is not 100% clear to me. I understand we now run them serial behind each other (the foreach), but why is the CatchupAsync now an infinite loop?

Reply:
The loop is normally in the `FeedProjectionRunner` which loops every x seconds over the atom feed to see if there are new changes. I removed that loop from `LinkedFeedProjectionRunner` and placed it on a higher level for these linked feeds.

So every x seconds all atom feeds will be checked for changes.